core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosClusterScheduler.scala

```diff
@@ -318,7 +320,7 @@ private[spark] class MesosClusterScheduler(
     ready = false
     metricsSystem.report()
     metricsSystem.stop()
-    mesosDriver.stop(true)
+    mesosDriver.stop(driverFailOver)
```

Comment:
Instead of killing without failover, we could also start it without failover.

In the `start` method, to use:
```scala
val driver = createSchedulerDriver(
       master,
       MesosClusterScheduler.this,
       Utils.getCurrentUserName(),
       appName,
       conf,
       Some(frameworkUrl),
       Some(driverFailOver),                                 // <-- with or without checkpoint data
       Some(if (driverFailOver) Double.MaxValue else 0.0),   // <-- timeout for failover recovery
       fwId)
```